app/assets/javascripts/admin/addon/components/emoji-value-list.js

```diff
@@ -0,0 +1,198 @@
+import Component from "@ember/component";
+import I18n from "I18n";
+import discourseComputed, { on } from "discourse-common/utils/decorators";
+import { emojiUrlFor } from "discourse/lib/text";
+import { action, set, setProperties } from "@ember/object";
+import { later, schedule } from "@ember/runloop";
+
+export default Component.extend({
+  classNameBindings: [":value-list", ":emoji-list"],
+  collection: null,
+  values: null,
+  validationMessage: null,
+  emojiPickerIsActive: false,
+  isEditorFocused: false,
+  emojiName: null,
+
+  init() {
+    this._super(...arguments);
+    this.set("collection", []);
+  },
+
+  @action
+  closeEmojiPicker() {
+    this.collection.setEach("isEditing", false);
+    this.set("emojiPickerIsActive", false);
+    this.set("isEditorFocused", false);
+  },
+
+  @action
+  emojiSelected(code) {
+    if (!this._validateInput(code)) {
+      return;
+    }
+
+    const item = this.collection.findBy("isEditing");
+    if (item) {
+      setProperties(item, {
+        value: code,
+        emojiUrl: emojiUrlFor(code),
+        isEditing: false,
+      });
+
+      this._saveValues();
+    } else {
+      this.set("emojiName", code);
+    }
+
+    this.set("emojiPickerIsActive", false);
+    this.set("isEditorFocused", false);
+  },
+
+  @action
+  clearInput() {
+    this.set("emojiName", null);
+  },
+
+  @on("didReceiveAttrs")
+  _setupCollection() {
+    this.set("collection", this._splitValues(this.values));
+  },
+
+  @discourseComputed("collection")
+  showUpDownButtons(collection) {
+    return collection.length - 1 ? true : false;
+  },
+
+  _splitValues(values) {
+    if (values && values.length) {
+      const emojiList = [];
+      const emojis = values.split("|").filter(Boolean);
+      emojis.forEach((emojiName) => {
+        const emoji = {
+          isEditable: true,
+          isEditing: false,
+        };
+        emoji.value = emojiName;
+        emoji.emojiUrl = emojiUrlFor(emojiName);
+
+        emojiList.push(emoji);
+      });
+
+      return emojiList;
+    } else {
+      return [];
+    }
+  },
+
+  @action
+  editValue(index) {
+    this.closeEmojiPicker();
+    schedule("afterRender", () => {
+      if (parseInt(index, 10) >= 0) {
+        const item = this.collection[index];
+        if (item.isEditable) {
+          set(item, "isEditing", true);
+        }
+      }
+
+      this.set("isEditorFocused", true);
+      later(() => {
+        if (!this.element || !this.isDestroyed || !this.isDestroying) {
```

Comment:
Your logic is incorrect:

```suggestion
        if (this.element && !this.isDestroying && !this.isDestroyed) {
```

Reply:
Oops surprised to see how it worked earlier on my local😞